src/main.rs

```diff
@@ -231,10 +227,11 @@ fn _main() -> PaziResult {
                     return PaziResult::Error;
                 }
             }
-        } else {
-            // unwrap is safe because of the 'matches.len() == 0' check above.
-            print!("{}", matches.last().unwrap().0);
+        } else if let Some((path, _)) = matches.next() {
+            print!("{}", path);
             res = PaziResult::SuccessDirectory;
+        } else {
+            res = PaziResult::Error;
```

Comment:
Previously this case had an immediate return vs setting res and falling through to doing a db save. Is this change intentional?

I do think there is a slight behavioural difference around jumps that only match trimmed directories (namely that they're kept longer with the immediate return).
I think that doing this, saving the database, is probably more correct, but I wanted to know if you had a similar intent or if this incidentally happened.

Reply:
Yep, that was the intent.

That said, I could see an argument for doing an immediate return here: if you try to jump to a path on removable media, but you've forgotten to insert the media first, doing an immediate return wouldn't lose that database entry.

Reply:
mmmm, that is a potentially nice property, though I think it ends up being too surprising.

If any other path matches, you jump there and it does clean up, but if no other path matches it doesn't? Yeah, seems like a weird rule.